src/graphics/canvas.rs

```diff
@@ -281,6 +283,26 @@ impl Canvas {
         self.screen
     }
 
+    /// Sets the scissor rectangle used when drawing. Nothing will be drawn to the canvas
+    /// that falls outside of this region.
+    ///
+    /// Note: The rectangle is in pixel coordinates, and therefore the values will be rounded towards zero.
+    #[inline]
+    pub fn set_scissor_rect(&mut self, rect: Rect) {
+        self.state.scissor_rect = (rect.x as u32, rect.y as u32, rect.w as u32, rect.h as u32);
+    }
+
+    /// Returns the scissor rectangle as set by [`Canvas::set_scissor_rect`].
+    #[inline]
+    pub fn scissor_rect(&self) -> Rect {
+        Rect::new(
+            self.state.scissor_rect.0 as f32,
+            self.state.scissor_rect.1 as f32,
+            self.state.scissor_rect.2 as f32,
+            self.state.scissor_rect.3 as f32,
+        )
```

Comment:
Why not simply copy it?

Reply:
scissor rect is stored as (u32, u32, u32, u32), not Rect - because scissor rects are in pixel coordinates

Reply:
Nevermind then, thought I checked it, but I guess I was too quick again.